packages/react/src/authenticate.js

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import { compose, lifecycle, withState, withProps } from 'recompose';
+import FormTypes from './FormTypes';
+
+const authenticate = ({
+  accounts,
+  Loading,
+  Dialog,
+}) => Component => compose(
+  withState('user', 'setUser', null),
+  withState('dialog', 'showDialog', false),
+  lifecycle({
+    async componentDidMount() {
+      this.store = accounts.instance.store;
+      this.user = null;
+      this.unsubscribe = this.store.subscribe(() => {
+        const user = this.store.getState().getIn(['accounts', 'user']);
+        if (user.size > 0 && !user.equals(this.user)) {
+          this.user = user;
+          this.props.setUser(user);
+        }
+      });
+      try {
+        this.props.showDialog(false);
+        await accounts.resumeSession();
+      } catch (err) {
+        this.props.showDialog(true);
+      }
+    },
+    componentWillUnmount() {
+      this.unsubscribe();
+    },
+  }),
+)(({ user, dialog }) => {
+  if (accounts.isLoading()) {
```

Comment:
actually when accounts.isLoading is false and user is empty render return null, instead it should redirect to login